### PR TITLE
Two moons test fit

### PR DESCRIPTION
**Pull request by Chase-Grajeda**
Adds tests to verify convergence in `test_two_moons/test_two_moons.py`

`test_fit`
- Tests for decreasing KL-Divergence loss after training
- Tests for decreasing validation loss after training
- Tests weights haven't vanished
- Tests model performance via MMD before and after training

### CODE REVIEW

File: tests/utils/ops.py [hunk 1]
```diff
@@ -7,3 +7,30 @@ def isclose(x1, x2, rtol=1e-5, atol=1e-8):
 
 def allclose(x1, x2, rtol=1e-5, atol=1e-8):
     return keras.ops.all(isclose(x1, x2, rtol, atol))
+
+
+def max_mean_discrepancy(x, y):
```
Comment (LarsKue) [inline tests/utils/ops.py:12]: In the future, let's put these in `bayesflow/metrics` directly. This time, I will move this and make it a bit more customizable.

File: tests/test_two_moons/test_two_moons.py [hunk 2]
```diff
@@ -11,14 +11,38 @@ def test_compile(approximator, random_samples, jit_compile):
...
-    )
+        epochs=3,
+    ).history
+    x_after = approximator.inference_network(inf_vars, conditions=inf_conds)
+    mmd_after = max_mean_discrepancy(x_after, y)
+
+    # Test model weights have not vanished
+    for layer in approximator.layers:
+        for weight in layer.weights:
+            assert not keras.ops.any(keras.ops.isnan(weight)).numpy()
```
Comment (LarsKue) [inline tests/test_two_moons/test_two_moons.py:38]: `tensor.numpy()` is not backend-agnostic and also not necessary here (when necessary, use `keras.ops.convert_to_numpy` instead)

File: tests/test_two_moons/test_two_moons.py [hunk 3]
```diff
@@ -11,14 +11,38 @@ def test_compile(approximator, random_samples, jit_compile):
...
+
+    approximator.compile(jit_compile=jit_compile, loss=keras.losses.KLDivergence())
+    inf_vars = approximator.configurator.configure_inference_variables(test_dataset.data)
+    inf_conds = approximator.configurator.configure_inference_conditions(test_dataset.data)
+    y = test_dataset.data["x"]
+
+    pre_loss = approximator.compute_metrics(train_dataset.data)["loss"]
+    pre_val_loss = approximator.compute_metrics(validation_dataset.data)["loss"]
+    x_before = approximator.inference_network(inf_vars, conditions=inf_conds)
+    mmd_before = max_mean_discrepancy(x_before, y)
```
Comment (LarsKue) [inline tests/test_two_moons/test_two_moons.py:25]: inconsistent naming

File: tests/test_two_moons/test_two_moons.py [hunk 4]
```diff
@@ -11,14 +11,38 @@ def test_compile(approximator, random_samples, jit_compile):
...
-    # TODO: verify the model learns something by comparing a metric before and after training
-    approximator.compile(jit_compile=jit_compile)
-    approximator.fit(
+def test_fit(approximator, train_dataset, validation_dataset, test_dataset, jit_compile):
+    # TODO: Refactor to use approximator.sample() when implemented (instead of calling the inference network directly)
+
+    approximator.compile(jit_compile=jit_compile, loss=keras.losses.KLDivergence())
+    inf_vars = approximator.configurator.configure_inference_variables(test_dataset.data)
+    inf_conds = approximator.configurator.configure_inference_conditions(test_dataset.data)
+    y = test_dataset.data["x"]
```
Comment (LarsKue) [inline tests/test_two_moons/test_two_moons.py:20]: relies on the internal structure of the dataset fixture, which is not good. Use `test_batch = test_dataset[0]; observables = test_batch["x"]` instead.

**Review by LarsKue — APPROVED** [review]
Great changes, thank you. I added some comments for future improvement, but I will merge this as-is. Also make sure to pass the linter check in the future (you can do this automatically by installing the pre-commit hook with `conda env update -f environment.yaml && pre-commit install && pre-commit run --all-files`.)
